lib/mof/helper.rb

```diff
@@ -80,15 +80,22 @@ def open name, origin = nil
     elsif bom == "\377\376"
       @iconv = "UTF-16LE"
       $/ = "\r\0\n\0"
-    elsif ! has_valid_utf8(@name)
-      $stderr.puts "#{name} contains invalid UTF-8, treating as ISO-8859-1"
-      @iconv = "ISO-8859-1"
-      @file.rewind
-      $/ = "\n"
     else
-      @file.rewind
-      @iconv = nil
-      $/ = "\n"
+      bom += @file.read(1)
+      # require 'pry'; binding.pry if @name =~ /^*.MSFT_xSPWebAppBlockedFileTypes\.schema\.mof$/
```

Comment:
Line left in for now so that others may easily debug the behavior on a bad file.